### ISSUE TITLE
`Error when subscribing a broadcast message` on tests. And crash when running in standalone mode

### ISSUE THREAD

**Issue by ASL07** [iotexproject/iotex-core#2577]
### What version of iotex-core are you using (`ioctl version`)?
Latest master branch

### What operating system and processor architecture are you using?
Processor: x86_64
OS: Ubuntu 20.04.1 LTS

### What did you do? If possible, provide a recipe for reproducing the error.
Followed steps on https://docs.iotex.io/developer/get-started/single-node.html for a build from source
Also tried a docker container following the section https://docs.iotex.io/developer/architecture-software.html#standalone-mode

### What did you expect to see?
Node runs correctly

### What did you see instead?
- Build is ok, but one test fails after running `make` with the following:
``2021-03-07T18:04:30.686Z        ERROR   go-p2p@v0.2.12/p2p.go:449       Error when subscribing a broadcast message.     {"error": "subscription cancelled by calling sub.Cancel()"}                                                                           
github.com/iotexproject/go-p2p.(*Host).AddBroadcastPubSub.func1``

- `make run` crashes witht the following:
```
go build -ldflags "-X 'github.com/iotexproject/iotex-core/pkg/version.PackageVersion=v1.1.3-31-g5c8f449f' -X 'github.com/iotexproject/iotex-core/pkg/version.PackageCommitID=5c8f449f8a6c8ce746d1c82870d760856985b241' -X 'github.com/iotexproject/iotex-core/pkg/version.GitStatus="dirty"' -X 'github.com/iotexproject/iotex-core/pkg/version.GoVersion=go version go1.16 linux/amd64' -X 'github.com/iotexproject/iotex-core/pkg/version.BuildTime=2021-03-07-GMT/18:40:07' -s -w" -o ./bin/server -v ./server
./bin/server -plugin=gateway -config-path=./config/standalone-config.yaml -genesis-path=./config/standalone-genesis.yaml
2021/03/07 18:40:08 maxprocs: Leaving GOMAXPROCS=6: CPU quota undefined
2021/03/07 18:40:08 Cannot config global logger, use default one:  open /var/log/s.log: permission denied
2021-03-07T18:40:08.852Z        INFO    server/main.go:65       Config in use: {Plugins:map[0:<nil>] Network:{Host:0.0.0.0 Port:4689 ExternalHost: ExternalPort:4689 BootstrapNodes:[] MasterKey:44a94e57552c1d38fb2e64882bcf234805961fa03d74f2212d4cb32d343dff6d RelayType: RateLimit:{GlobalUnicastAvg:300 GlobalUnicastBurst:500 PeerAvg:300 PeerBurst:500} EnableRateLimit:true PrivateNetworkPSK:} Chain:{ChainDBPath:/var/data/chain.db TrieDBPath:/var/data/trie.db IndexDBPath:/var/data/index.db BloomfilterIndexDBPath:/var/data/bloomfilter.index.db CandidateIndexDBPath:/var/data/candidate.index.db StakingIndexDBPath:/var/data/staking.index.db ID:1 Address: ProducerPrivKey: SignatureScheme:[secp256k1] EmptyGenesis:false GravityChainDB:{DbPath:/var/data/poll.db NumRetries:10 MaxCacheSize:0 BlockStoreBatchSize:0 V2BlocksToSplitDB:0 Compressor: CompressLegacy:false SplitDBSizeMB:0 SplitDBHeight:0 HistoryStateRetention:0} Committee:{NumOfRetries:0 GravityChainAPIs:[] GravityChainHeightInterval:0 GravityChainStartHeight:0 GravityChainCeilingHeight:0 RegisterContractAddress: StakingContractAddress: PaginationSize:0 VoteThreshold: ScoreThreshold: SelfStakingThreshold: CacheSize:0 NumOfFetchInParallel:0 SkipManifiedCandidate:false GravityChainBatchSize:0} EnableTrielessStateDB:true EnableStateDBCaching:false EnableArchiveMode:false EnableAsyncIndexWrite:true EnableSystemLogIndexer:false EnableStakingProtocol:true EnableStakingIndexer:false CompressBlock:false AllowedBlockGasResidue:10000 MaxCacheSize:0 PollInitialCandidatesInterval:10s StateDBCacheSize:1000 WorkingSetCacheSize:20} ActPool:{MaxNumActsPerPool:32000 MaxGasLimitPerPool:320000000 MaxNumActsPerAcct:2000 ActionExpiry:10m0s MinGasPriceStr:1000000000000 BlackList:[]} Consensus:{Scheme:STANDALONE RollDPoS:{FSM:{EventChanSize:10000 UnmatchedEventTTL:3s UnmatchedEventInterval:100ms AcceptBlockTTL:4s AcceptProposalEndorsementTTL:2s AcceptLockEndorsementTTL:2s CommitTTL:2s} ToleratedOvertime:2s Delay:5s ConsensusDBPath:/var/data/consensus.db}} BlockSync:{Interval:10s ProcessSyncRequestTTL:10s BufferSize:200 IntervalSize:20 MaxRepeat:3 RepeatDecayStep:1} Dispatcher:{EventChanSize:10000} API:{UseRDS:false Port:14014 TpsWindow:10 GasStation:{SuggestBlockWindow:20 DefaultGas:1000000000000 Percentile:60} RangeQueryLimit:1000} System:{Active:true HeartbeatInterval:10s HTTPAdminPort:9009 HTTPStatsPort:8080 StartSubChainInterval:10s SystemLogDBPath:/var/log/systemlog.db} DB:{DbPath: NumRetries:3 MaxCacheSize:64 BlockStoreBatchSize:16 V2BlocksToSplitDB:1000000 Compressor:Snappy CompressLegacy:false SplitDBSizeMB:0 SplitDBHeight:900000 HistoryStateRetention:2000} Indexer:{RangeBloomFilterNumElements:100000 RangeBloomFilterSize:1200000 RangeBloomFilterNumHash:8} Log:{Zap:0xc0000df200 StderrRedirectFile:0xc000512540 RedirectStdLog:true} SubLogs:map[global:{Zap:0xc0000df200 StderrRedirectFile:0xc000512540 RedirectStdLog:true}] Genesis:{Blockchain:{Timestamp:1546329600 BlockGasLimit:20000000 ActionGasLimit:5000000 BlockInterval:5s NumSubEpochs:15 DardanellesNumSubEpochs:30 NumDelegates:24 NumCandidateDelegates:36 TimeBasedRotation:true PacificBlockHeight:432001 AleutianBlockHeight:864001 BeringBlockHeight:1512001 CookBlockHeight:1641601 DardanellesBlockHeight:1816201 DaytonaBlockHeight:3238921 EasterBlockHeight:4478761 FbkMigrationBlockHeight:5157001 FairbankBlockHeight:5165641 GreenlandBlockHeight:6544441 HawaiiBlockHeight:11073241} Account:{InitBalanceMap:map[io1zzaqgns8t3zg97502mq84z5jg54m2mfftgsua6:100000000000000000000000000000000000]} Poll:{PollMode:nativeMix EnableGravityChainVoting:true GravityChainStartHeight:0 GravityChainCeilingHeight:10199000 GravityChainHeightInterval:0 RegisterContractAddress: StakingContractAddress: NativeStakingContractAddress: NativeStakingContractCode: ConsortiumCommitteeContractCode: VoteThreshold: ScoreThreshold: SelfStakingThreshold: Delegates:[] ProbationEpochPeriod:6 ProbationIntensityRate:90 UnproductiveDelegateMaxCacheSize:20} Rewarding:{InitBalanceStr:200000000000000000000000000 BlockRewardStr:16000000000000000000 DardanellesBlockRewardStr:8000000000000000000 EpochRewardStr:12500000000000000000000 AleutianEpochRewardStr:18750000000000000000000 NumDelegatesForEpochReward:100 ExemptAddrStrsFromEpochReward:[] FoundationBonusStr:80000000000000000000 NumDelegatesForFoundationBonus:36 FoundationBonusLastEpoch:8760 FoundationBonusP2StartEpoch:9698 FoundationBonusP2EndEpoch:18458 ProductivityThreshold:0} Staking:{VoteWeightCalConsts:{DurationLg:1.2 AutoStake:1 SelfStake:1.06} RegistrationConsts:{Fee:100000000000000000000 MinSelfStake:1200000000000000000000000} WithdrawWaitingPeriod:72h0m0s MinStakeAmount:100000000000000000000 BootstrapCandidates:[]}}}
2021-03-07T18:40:08.852Z        INFO    blockchain/blockchain.go:408    Add a subscriber.
2021-03-07T18:40:08.852Z        INFO    blockchain/blockchain.go:408    Add a subscriber.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xfb2336]

goroutine 1 [running]:
github.com/iotexproject/iotex-core/chainservice.New(0xc000538000, 0xc00011bc78, 0x7, 0x1251, 0x0, 0x0, 0x1251, 0x1eeb1f0, 0x0, 0x0, ...)
        /home/adri/iotex/iotex-core/chainservice/chainservice.go:388 +0xfd6
github.com/iotexproject/iotex-core/server/itx.newServer(0xc000538000, 0xc00011bc78, 0x7, 0x1251, 0x0, 0x0, 0x1251, 0x1eeb1f0, 0x0, 0x0, ...)
        /home/adri/iotex/iotex-core/server/itx/server.go:69 +0x2fc
github.com/iotexproject/iotex-core/server/itx.NewServer(...)
        /home/adri/iotex/iotex-core/server/itx/server.go:46
main.main()
        /home/adri/iotex/iotex-core/server/main.go:86 +0x8c8
make: *** [Makefile:183: run] Error 2
```

- Running on docker also crashes with the following:
```
sudo docker run -a stdin -a stderr -a stdout -p 30100:14004 --mount type=bind,source=$PWD/standalone-config.yaml,target=/etc/iotex/config_local_delegate.yaml iotex/iotex-core:latest iotex-server -config-path=/etc/iotex/config_local_delegate.yaml
2021/03/07 18:37:05 maxprocs: Leaving GOMAXPROCS=6: CPU quota undefined
{"level":"info","ts":1615142225.852707,"caller":"server/main.go:65","msg":"Config in use: {Plugins:map[] Network:{Host:0.0.0.0 Port:4689 ExternalHost: ExternalPort:4689 BootstrapNodes:[] MasterKey:c8d84b6acc33811cb70e3f8b4eb6eaa98747d61e6fd079bb219e92f9d6ab6727 RelayType: RateLimit:{GlobalUnicastAvg:300 GlobalUnicastBurst:500 PeerAvg:300 PeerBurst:500} EnableRateLimit:true PrivateNetworkPSK:} Chain:{ChainDBPath:/var/data/chain.db TrieDBPath:/var/data/trie.db IndexDBPath:/var/data/index.db BloomfilterIndexDBPath:/var/data/bloomfilter.index.db CandidateIndexDBPath:/var/data/candidate.index.db StakingIndexDBPath:/var/data/staking.index.db ID:1 Address: ProducerPrivKey: SignatureScheme:[secp256k1] EmptyGenesis:false GravityChainDB:{DbPath:/var/data/poll.db NumRetries:10 MaxCacheSize:0 BlockStoreBatchSize:0 V2BlocksToSplitDB:0 Compressor: CompressLegacy:false SplitDBSizeMB:0 SplitDBHeight:0 HistoryStateRetention:0} Committee:{NumOfRetries:0 GravityChainAPIs:[] GravityChainHeightInterval:0 GravityChainStartHeight:0 GravityChainCeilingHeight:0 RegisterContractAddress: StakingContractAddress: PaginationSize:0 VoteThreshold: ScoreThreshold: SelfStakingThreshold: CacheSize:0 NumOfFetchInParallel:0 SkipManifiedCandidate:false GravityChainBatchSize:0} EnableTrielessStateDB:true EnableStateDBCaching:false EnableArchiveMode:false EnableAsyncIndexWrite:true EnableSystemLogIndexer:false EnableStakingProtocol:true EnableStakingIndexer:false CompressBlock:false AllowedBlockGasResidue:10000 MaxCacheSize:0 PollInitialCandidatesInterval:10s StateDBCacheSize:1000 WorkingSetCacheSize:20} ActPool:{MaxNumActsPerPool:32000 MaxGasLimitPerPool:320000000 MaxNumActsPerAcct:2000 ActionExpiry:10m0s MinGasPriceStr:1000000000000 BlackList:[]} Consensus:{Scheme:STANDALONE RollDPoS:{FSM:{EventChanSize:10000 UnmatchedEventTTL:3s UnmatchedEventInterval:100ms AcceptBlockTTL:4s AcceptProposalEndorsementTTL:2s AcceptLockEndorsementTTL:2s CommitTTL:2s} ToleratedOvertime:2s Delay:5s ConsensusDBPath:/var/data/consensus.db}} BlockSync:{Interval:10s ProcessSyncRequestTTL:10s BufferSize:200 IntervalSize:20 MaxRepeat:3 RepeatDecayStep:1} Dispatcher:{EventChanSize:10000} API:{UseRDS:false Port:14014 TpsWindow:10 GasStation:{SuggestBlockWindow:20 DefaultGas:1000000000000 Percentile:60} RangeQueryLimit:1000} System:{Active:true HeartbeatInterval:10s HTTPAdminPort:9009 HTTPStatsPort:8080 StartSubChainInterval:10s SystemLogDBPath:/var/log/systemlog.db} DB:{DbPath: NumRetries:3 MaxCacheSize:64 BlockStoreBatchSize:16 V2BlocksToSplitDB:1000000 Compressor:Snappy CompressLegacy:false SplitDBSizeMB:0 SplitDBHeight:900000 HistoryStateRetention:2000} Indexer:{RangeBloomFilterNumElements:100000 RangeBloomFilterSize:1200000 RangeBloomFilterNumHash:8} Log:{Zap:0xc0006e4000 StderrRedirectFile:0xc0006d70e0 RedirectStdLog:true} SubLogs:map[global:{Zap:0xc0006e4000 StderrRedirectFile:0xc0006d70e0 RedirectStdLog:true}] Genesis:{Blockchain:{Timestamp:1546329600 BlockGasLimit:20000000 ActionGasLimit:5000000 BlockInterval:10s NumSubEpochs:2 DardanellesNumSubEpochs:30 NumDelegates:24 NumCandidateDelegates:36 TimeBasedRotation:false PacificBlockHeight:432001 AleutianBlockHeight:864001 BeringBlockHeight:1512001 CookBlockHeight:1641601 DardanellesBlockHeight:1816201 DaytonaBlockHeight:3238921 EasterBlockHeight:4478761 FbkMigrationBlockHeight:5157001 FairbankBlockHeight:5165641 GreenlandBlockHeight:6544441 HawaiiBlockHeight:11073241} Account:{InitBalanceMap:map[]} Poll:{PollMode:nativeMix EnableGravityChainVoting:true GravityChainStartHeight:0 GravityChainCeilingHeight:10199000 GravityChainHeightInterval:0 RegisterContractAddress: StakingContractAddress: NativeStakingContractAddress: NativeStakingContractCode: ConsortiumCommitteeContractCode: VoteThreshold: ScoreThreshold: SelfStakingThreshold: Delegates:[] ProbationEpochPeriod:6 ProbationIntensityRate:90 UnproductiveDelegateMaxCacheSize:20} Rewarding:{InitBalanceStr:200000000000000000000000000 BlockRewardStr:16000000000000000000 DardanellesBlockRewardStr:8000000000000000000 EpochRewardStr:12500000000000000000000 AleutianEpochRewardStr:18750000000000000000000 NumDelegatesForEpochReward:100 ExemptAddrStrsFromEpochReward:[] FoundationBonusStr:80000000000000000000 NumDelegatesForFoundationBonus:36 FoundationBonusLastEpoch:8760 FoundationBonusP2StartEpoch:9698 FoundationBonusP2EndEpoch:18458 ProductivityThreshold:0} Staking:{VoteWeightCalConsts:{DurationLg:1.2 AutoStake:1 SelfStake:1.06} RegistrationConsts:{Fee:100000000000000000000 MinSelfStake:1200000000000000000000000} WithdrawWaitingPeriod:72h0m0s MinStakeAmount:100000000000000000000 BootstrapCandidates:[]}}}","ioAddr":"io1f7qsjtf7efjupqrxpxmhgejpjerhjt90mxlum3","networkAddr":"0.0.0.0:4689"}
{"level":"info","ts":1615142225.853873,"caller":"blockchain/blockchain.go:408","msg":"Add a subscriber.","ioAddr":"io1f7qsjtf7efjupqrxpxmhgejpjerhjt90mxlum3","networkAddr":"0.0.0.0:4689"}
```

**Comment by millken** [comment]
The default standalone configuration requires root privileges,
try `sudo ./bin/server -plugin=gateway -config-path=./config/standalone-config.yaml -genesis-path=./config/standalone-genesis.yaml
`

**Comment by millken** [comment]
fixed in #2579
